--- origsrc/date.c	2014-08-25 21:41:21.000000000 -0500
+++ src/date.c	2014-10-13 20:58:30.770502300 -0500
@@ -348,12 +348,19 @@ reset(time_t newt, bool nflag)
 		struct utmpx	after;
 	} sx;
 #endif
+#ifdef __CYGWIN__
+	struct timeval		tv = { newt, 0 };
+#endif
 
 	/*
 	** Wouldn't it be great if stime returned the old time?
 	*/
 	oldt = time(NULL);
+#ifdef __CYGWIN__
+	if (settimeofday(&tv, NULL) != 0)
+#else
 	if (stime(&newt) != 0)
+#endif
 		oops("stime");
 	s.before.ut_type = OLD_TIME;
 	s.before.ut_time = oldt;
